buildtools/aho-corasick/src/lib.rs

```diff
@@ -35,25 +33,15 @@ pub extern "C" fn new_matcher(patterns_ptr: *mut u8, patterns_len: usize) -> usi
         .match_kind(MatchKind::LeftmostLongest)
         .build(patterns);
 
-    unsafe {
-        MATCHERS.push(ac);
-        MATCHERS.len() - 1
-    }
-
+    return Box::new(ac)
 }
 
 #[no_mangle]
-pub extern "C" fn matches(matcher_ptr: usize, value_ptr: usize, value_len: usize, n: usize, matches: *mut usize) -> usize {
-    let ac = unsafe {
-        let matcher = MATCHERS.get_unchecked(matcher_ptr);
-        matcher
-    };
-
+pub extern "C" fn matches(ac: &mut AhoCorasick, value_ptr: usize, value_len: usize, n: usize, matches: *mut usize) -> usize {
     let value = ptr_to_string(value_ptr, value_len);
-    std::mem::forget(&value);
```

Comment:
The intention was to wrap with `String` and forget it to prevent deallocating the passed in string from Rust. But I think this was just leaking a Rust allocated / copied string. Not sure though since don't understand Rust that well